src/irmin-pack/unix/gc_worker.ml

```diff
@@ -278,10 +277,11 @@ module Make (Args : Gc_args.S) = struct
           |> Errs.log_if_error "GC: Close prefix after parent rewrite")
       @@ fun () ->
       let write_exn = Io.write_exn prefix in
-      List.iter
-        (fun key ->
-          transfer_parent_commit_exn ~read_exn ~write_exn ~mapping key)
-        (Commit_value.parents commit)
+      if not !indexed_parent then
```

Comment:
Previously, we would raise an exception for a parent commit key that is indexed but now we ignore it. My only concern is that what used to cause noise (an exception) will now skip silently. But perhaps this is okay. If so, I can imagine future readers might be confused about this logic. Could you add a small comment somewhere to explain why this is safe to do / the context we expect it to happen?

Reply:
I expanded on the comment above, where we explain why we need to include the parent commit in the iteration. Raising an exception in this case, was I think, just by simplicity, since we assumed that the gc is always called on stores that use minimal indexing strategy. The case of the (old) snapshot import is a bit weird, the parent commit is referenced by hash, however, it is not in the index. 

Let me know if that comment look ok.

Reply:
> Hmm, looks like your changes didn't get pushed.

Ah, yes, sorry about that.

> so this code is for handling these (older?) versions of commits that used hashes instead of offsets.

Yes, exactly. 

> Do we want to block GC-based snapshotting for these stores? Can we? Or is it just an edge case we have to handle? We already block GC for non-minimal indexed stores, so I guess it is more than just that if we wanted to block them.

You are right, we cannot do the gc-based snapshot on non-minimal indexed stores. The gc will fail when inspecting the key of nodes and contents that are indexed (https://github.com/mirage/irmin/blob/main/src/irmin-pack/unix/gc_worker.ml#L215). We need to make an exception for parent commits, in order to (temporary) support the older snapshots.

Reply:
Also, I realised with your comment that the code could handle cases where some of the parent commits are referenced by hash and some by offset. So I change it to handle this case as well.